spec/github_helper_spec.rb

```diff
@@ -171,4 +133,151 @@ def mock_comment(body: '<!-- REUSE_ID: test-id --> Test', user_id: 1234)
       instance_double('Comment', id: 1234, body: body, user: instance_double('User', id: user_id))
     end
   end
+
+  describe 'get_milestone' do
+    let(:test_repo) { 'repo-test/project-test' }
+    let(:test_milestone) { [{ title: 'release/10.0' }, { title: 'release/10.1' }, { title: 'hotfix/10.2' }] }
+    let(:client) do
+      instance_double(
+        Octokit::Client,
+        list_milestones: []
+      )
+    end
+
+    before do
+      allow(described_class).to receive(:client).and_return(client)
+    end
+
+    it 'properly passes the repository all the way down to the Octokit::Client' do
+      expect(client).to receive(:list_milestones).with(test_repo)
+      described_class.get_milestone(test_repo, 'test')
+    end
+
+    it 'returns nil when no milestone is returned from the api' do
+      expect(described_class.get_milestone(test_repo, 'release')).to be_nil
+    end
+
+    it 'returns nil when no milestone title starts with the searched term' do
+      allow(client).to receive(:list_milestones).and_return(test_milestone)
+      expect(described_class.get_milestone(test_repo, '10.0')).to be_nil
+    end
+
+    it 'returns a milestone when the milestone title starts with search term' do
+      allow(client).to receive(:list_milestones).and_return(test_milestone)
+      expect(described_class.get_milestone(test_repo, 'hotfix')).to eq({ title: 'hotfix/10.2' })
+    end
+
+    it 'returns the newest of milestones where the title matches with search term' do
+      allow(client).to receive(:list_milestones).and_return(test_milestone)
+      expect(described_class.get_milestone(test_repo, 'release')).to eq({ title: 'release/10.1' })
+    end
```

Comment:
Self-Review: I'm not sure if those titles reflect the purpose of the unit test well, but I can't think of any other way to describe it right now (Naming is hard)

Reply:
Those looks great spec titles to me 👍